File: call-management/business-hours/update-user-business-hours/code-samples/updateUserBusinessHours.php
<?php
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
$accountId = '<ENTER VALUE>';
$extensionId = '<ENTER VALUE>';

// POST BODY
$body = array(
    'schedule' => array(
        'weeklyRanges' => array(
            'monday' => array(
                array(
                    'from' => '<ENTER VALUE>',
                    'to' => '<ENTER VALUE>'
                )  
            ),
            'tuesday' => array(
                array(
                    'from' => '<ENTER VALUE>',
                    'to' => '<ENTER VALUE>'
                )  
            ),
            'wednesday' => array(
                array(
                    'from' => '<ENTER VALUE>',
                    'to' => '<ENTER VALUE>'
                )  
            ),
            'thursday' => array(
                array(
                    'from' => '<ENTER VALUE>',
                    'to' => '<ENTER VALUE>'
                )  
            ),
            'friday' => array(
                array(
                    'from' => '<ENTER VALUE>',
                    'to' => '<ENTER VALUE>'
                )  
            ),
            'saturday' => array(
                array(
                    'from' => '<ENTER VALUE>',
                    'to' => '<ENTER VALUE>'
                )  
            ),
            'sunday' => array(
                array(
                    'from' => '<ENTER VALUE>',
                    'to' => '<ENTER VALUE>'
                )  
            )
        )
    )
);

require('vendor/autoload.php');
$rcsdk = new RingCentral\SDK\SDK(getenv('clientId'), getenv('clientSecret'), getenv('serverURL'));
$platform = $rcsdk->platform();
$platform->login(getenv('username'), getenv('extension'), getenv('password'));
$r = $platform->put("/restapi/v1.0/account/{$accountId}/extension/{$extensionId}/business-hours", $body);
// PROCESS RESPONSE
?>

File: call-management/business-hours/update-user-business-hours/code-samples/updateUserBusinessHours.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// PATH PARAMETERS
string accountId = "<ENTER VALUE>";
string extensionId = "<ENTER VALUE>";

// POST BODY
UserBusinessHoursUpdateRequest userBusinessHoursUpdateRequest = new UserBusinessHoursUpdateRequest {
    schedule = new UserBusinessHoursScheduleInfo {
        weeklyRanges = new WeeklyScheduleInfo {
            monday = new[] {
                new TimeInterval {
                    from = "<ENTER VALUE>",
                    to = "<ENTER VALUE>"
                },
            },
            tuesday = new[] {
                new TimeInterval {
                    from = "<ENTER VALUE>",
                    to = "<ENTER VALUE>"
                },
            },
            wednesday = new[] {
                new TimeInterval {
                    from = "<ENTER VALUE>",
                    to = "<ENTER VALUE>"
                },
            },
            thursday = new[] {
                new TimeInterval {
                    from = "<ENTER VALUE>",
                    to = "<ENTER VALUE>"
                },
            },
            friday = new[] {
                new TimeInterval {
                    from = "<ENTER VALUE>",
                    to = "<ENTER VALUE>"
                },
            },
            saturday = new[] {
                new TimeInterval {
                    from = "<ENTER VALUE>",
                    to = "<ENTER VALUE>"
                },
            },
            sunday = new[] {
                new TimeInterval {
                    from = "<ENTER VALUE>",
                    to = "<ENTER VALUE>"
                },
            }
        }
    }
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Restapi().Account(accountId).Extension(extensionId).BusinessHours().Put(userBusinessHoursUpdateRequest);
// PROCESS RESPONSE
